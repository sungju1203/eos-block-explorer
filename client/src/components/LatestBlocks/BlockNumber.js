import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

import { fromTheme } from "helper"

const Wrapper = styled.div`
  background: ${fromTheme('primary')};
  color: ${fromTheme('white')};
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 80px;
  width: 150px;
`

const IdWrapper = styled.div`
  width: 100px;
  font-size: 10px;
  margin-top: 3px;
  overflow: hidden;
  text-overflow: ellipsis;
`

const BlockNumber = ({ blockNumber, id }) => (
  <Wrapper>
    <div> Block {blockNumber} </div>
    <IdWrapper> {id} </IdWrapper>
  </Wrapper>
)

BlockNumber.propTypes = {
  blockNumber: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
}

export default BlockNumber