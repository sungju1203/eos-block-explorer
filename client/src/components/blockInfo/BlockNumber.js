import React from "react"
import styled from "styled-components"

const Wrapper = styled.div`
  background: ${props => props.theme.primary};
  color: ${props => props.theme.white};
  display: flex;
  justify-content: center;
  align-items: center;
  height: 5em;
  width: 10em;
`

const BlockNumber = ({ blockNumber }) => (
  <Wrapper>
    Block {blockNumber}
  </Wrapper>
)

export default BlockNumber