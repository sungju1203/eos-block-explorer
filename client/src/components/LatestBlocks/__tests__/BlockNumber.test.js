import React from "react"
import renderer from "react-test-renderer"

import BlockNumber from "../BlockNumber"


test('Block Number', () => {
  const props = {
    blockNumber: 1437576,
    id: '0015ef887c1badff4d207fc03af2ceb78f935ba380e45b71b601901ddb7e508d',
  }

  const component = renderer.create(
    <BlockNumber {...props} />,
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})