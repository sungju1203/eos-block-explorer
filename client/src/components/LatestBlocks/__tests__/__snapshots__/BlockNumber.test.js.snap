// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block Number 1`] = `
<div
  className="sc-bdVaJa jGVUjf"
>
  <div>
     Block 
    1437576
     
  </div>
  <div
    className="sc-bwzfXH jPuMXn"
  >
     
    0015ef887c1badff4d207fc03af2ceb78f935ba380e45b71b601901ddb7e508d
     
  </div>
</div>
`;
